backend/de.metas.contracts/src/main/java/de/metas/contracts/impl/SubscriptionTermEventListener.java

```diff
@@ -98,6 +100,12 @@ else if (X_C_Flatrate_Conditions.ONFLATRATETERMEXTEND_CopyPrice.equals(condition
 			next.setC_TaxCategory_ID(predecessor.getC_TaxCategory_ID());
 			next.setIsTaxIncluded(predecessor.isTaxIncluded());
 		}
+		else if (ONFLATRATETERMEXTEND_ExtensionNotAllowed.equals(conditions.getOnFlatrateTermExtend()))
+		{
+			throw new AdempiereException("Extension Not Allowed")
```

Comment:
Should probably create a dedicated AD_Message entry for this, if this error message is displayed to the UI.